docs/core/extensions/dependency-injection-guidelines.md

```diff
@@ -0,0 +1,161 @@
+---
+title: Dependency injection guidelines
+description: Learn various dependency injection guidelines and best practices for .NET application development.
+author: IEvangelist
+ms.author: dapine
+ms.date: 09/23/2020
+ms.topic: guide
+---
+
+# Dependency injection guidelines
+
+In this article, you'll learn general guidelines and best practices for implementing dependency injection in your .NET applications. From design principles to proper disposal patterns, thread safety, and service lifetime management - these recommendations are tried and true.
+
+## Design services for dependency injection
+
+When designing services for dependency injection:
+
+- Avoid stateful, static classes and members. Avoid creating global state by designing apps to use singleton services instead.
+- Avoid direct instantiation of dependent classes within services. Direct instantiation couples the code to a particular implementation.
+- Make services small, well-factored, and easily tested.
+
+If a class has many injected dependencies, it might be a sign that the class has too many responsibilities and violates the [Single Responsibility Principle (SRP)](/dotnet/standard/modern-web-apps-azure-architecture/architectural-principles#single-responsibility). Attempt to refactor the class by moving some of its responsibilities into new classes. Keep in mind that Razor Pages page model classes and MVC controller classes should focus on UI concerns.
```

Comment:
> Keep in mind that Razor Pages page model classes and MVC controller classes should focus on UI concerns.

Sounds like a comment for the ASP.NET Core edition of DI docs.

```suggestion
If a class has many injected dependencies, it might be a sign that the class has too many responsibilities and violates the [Single Responsibility Principle (SRP)](/dotnet/standard/modern-web-apps-azure-architecture/architectural-principles#single-responsibility). Attempt to refactor the class by moving some of its responsibilities into new classes.
```